packages/circuit-ui/components/ButtonGroup/ButtonGroup.tsx

```diff
@@ -90,21 +99,78 @@ const inlineMobileStyles = ({
     ${getInlineStyles({ theme })}
   `;
 
+const actionStyles = ({ theme }: StyleProps) => css`
+  display: flex;
+  flex-wrap: wrap;
+  justify-content: center;
+  width: 100%;
+
+  > button,
+  > a {
+    width: auto;
+
+    &:not(:last-child) {
+      margin-right: ${theme.spacings.mega};
+      margin-bottom: 0;
+    }
+  }
+
+  ${theme.mq.untilKilo} {
```

Comment:
We decided to go with option 1. JS in the end by using `useMedia`. We tried different ways with CSS but it turns out to be too hacky and complicated. @connor-baer you can check it here [098d7cd](https://github.com/sumup-oss/circuit-ui/pull/1167/commits/098d7cd4dfd5cb11c1a166b4d8f4bef1b5eac81d) and let us know what you think about it.

Reply:
The problem with `useMedia` and any other JS-based solution is that it causes a layout shift/flicker of styles when doing server-side rendering. In my opinion, that's a deal-breaker. I would also favor a CSS-only approach.

Reply:
☝️ completely agree on principle, so we gave it a shot but it's not very clean:

- the styles we want are in the Button's `tertiaryStyles` style function, which use the `Button` props (including `variant`, `destructive` etc.) to render the right styles. If we import it, we'd need to forward it these props. Was this what you were thinking? (this was my initial idea but the implementation doesn't look great)
- alternatively we can copy the styles over, but this would duplicate them, which isn't so nice either

Reply:
Amela and I just paired on this and came up with a third approach: We render 3 buttons (secondary, primary, and tertiary). The secondary and tertiary are hidden on mobile and desktop respectively so only two buttons are visible at a time. 

This way, we don't need to override the `variant` prop or the variant styles. The main benefit, however, is that the tab order is correct on all viewports.

Reply:
It's a really good idea! I see you also tweaked the flex wrapper to change direction on the MQ. This was my main concern since the breakpoint would have been inconsistent if it were based on button width.

I liked the flex-wrap, but this MQ looks like the only way to make the breakpoint work well with switching the variants. Also it makes the breakpoint consistent if e.g. there are several ButtonGroups on the same page with different button widths